pkg/fakedri/fakedri.go

```diff
@@ -0,0 +1,529 @@
+
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+
+	opts.files++
+
+	node := 0
+	if opts.DevsPerNode > 0 {
+		node = i / opts.DevsPerNode
+	}
+
+	data = []byte(strconv.Itoa(node))
+	file = filepath.Join(base, "device", "numa_node")
+
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+
+	opts.files++
+
+	if opts.VfsPerPf > 0 && i%(opts.VfsPerPf+1) == 0 {
+		data = []byte(strconv.Itoa(opts.VfsPerPf))
+		file = filepath.Join(base, "device", "sriov_numvfs")
+
+		if err := os.WriteFile(file, data, fileMode); err != nil {
+			return err
+		}
+
+		opts.files++
+	}
+
+	for tile := 0; tile < opts.TilesPerDev; tile++ {
+		path := filepath.Join(base, "gt", fmt.Sprintf("gt%d", tile))
+		if err := os.MkdirAll(path, dirMode); err != nil {
+			return err
+		}
+
+		opts.dirs++
+	}
+
+	return nil
+}
+
+func addSysfsBusTree(root string, opts *GenOptions, i int) error {
+	pciName := fmt.Sprintf("0000:00:0%d.0", i)
+	base := filepath.Join(root, "bus", "pci", "drivers", opts.Driver, pciName)
+
+	if err := os.MkdirAll(base, dirMode); err != nil {
+		return err
+	}
+
+	opts.dirs++
+
+	data := []byte("0x4905")
+	file := filepath.Join(base, "device")
+
+	if err := os.WriteFile(file, data, fileMode); err != nil {
+		return err
+	}
+
+	opts.files++
+
+	drm := filepath.Join(base, "drm")
+	if err := os.MkdirAll(drm, dirMode); err != nil {
+		return err
+	}
+
+	opts.dirs++
+
+	return addDeviceNodes(drm, opts, i)
+}
+
+func addDeviceNodes(base string, opts *GenOptions, i int) error {
+	mode := uint32(fileMode | devNullType)
+	devid := int(unix.Mkdev(uint32(devNullMajor), uint32(devNullMinor)))
+
+	file := filepath.Join(base, fmt.Sprintf("card%d", cardBase+i))
+	if err := unix.Mknod(file, mode, devid); err != nil {
+		klog.Fatalf("NULL device (%d:%d) node creation failed for '%s': %v",
+			devNullMajor, devNullMinor, file, err)
+	}
+
+	opts.devs++
+
+	file = filepath.Join(base, fmt.Sprintf("renderD%d", renderBase+i))
+	if err := unix.Mknod(file, mode, devid); err != nil {
+		klog.Fatalf("NULL device (%d:%d) node creation failed for '%s': %v",
+			devNullMajor, devNullMinor, file, err)
+	}
+
+	opts.devs++
+
+	return nil
+}
+
+func addDeviceSymlinks(base string, opts *GenOptions, i int) error {
+	target := filepath.Join(base, fmt.Sprintf("by-path/pci-0000:%02d:02.0-card", i))
+	if err := os.Symlink(fmt.Sprintf("../card%d", cardBase+i), target); err != nil {
+		klog.Fatalf("symlink creation failed '%s': %v",
+			target, err)
+	}
+
+	opts.symls++
+
+	target = filepath.Join(base, fmt.Sprintf("by-path/pci-0000:%02d:02.0-render", i))
+	if err := os.Symlink(fmt.Sprintf("../renderD%d", renderBase+i), target); err != nil {
+		klog.Fatalf("symlink creation failed '%s': %v",
+			target, err)
+	}
+
+	opts.symls++
+
+	return nil
+}
+
+func addDevfsDriTree(root string, opts *GenOptions, i int) error {
+	base := filepath.Join(root, "dri")
+	if err := os.MkdirAll(base, dirMode); err != nil {
+		return err
+	}
+
```

Comment:
Redundant with the line below:
```suggestion
	if err := os.MkdirAll(base, dirMode); err != nil {
		return err
	}

```

Reply:
removed